src/govuk/components/character-count/character-count.mjs

```diff
@@ -1,7 +1,8 @@
 import '../../vendor/polyfills/Function/prototype/bind.mjs'
 import '../../vendor/polyfills/Event.mjs' // addEventListener and event.target normalisation
 import '../../vendor/polyfills/Element/prototype/classList.mjs'
```

Comment:
Do we put the polyfill here?

```suggestion
import '../../vendor/polyfills/Element/prototype/classList.mjs'
import '../../vendor/polyfills/Element/prototype/closest.mjs'
```

Reply:
If we move it here, we'd have to move the `closestAttributeValue` function as well as it's the one requiring the polyfill. We need to export it for testing, though and I think Rollup will complain about mixing named and not named exports, I think. It could be worth extracting the method and its polyfill import in its own file (`src/closestAttributeValue.mjs` for ex.) to avoid burdening individual imports of other components with the polyfill, that's a good point.

Reply:
Oh, I see, good point about polyfills being side effects (thanks for the detailed explanation, as always).

I'm not a huge fan of putting them in each components as a blanket rule, but more in "whatever requires the polyfill". That'll limit the risks of a component loading, say `closestAttributeValue` (or `normaliseString`) and forgetting to polyfill. With this approach we would:

- make a separate module for `closestAttributeValue` that imports `Element.prototype.closest`
- make a separate module for `normaliseString` that imports `String.prototype.trim`
- import `Element.prototype.dataset` in all components requiring configuration as they access the `dataset`. Given that one will be an easy one to miss, I wonder if we shouldn't have a `getDataset` function that has side effect.

These could live in a `polyfilled` folder to be clear about why those methods are on their own and we could revisit at regular interval (starting with 5.0) the need for them.

All in all, I think that's something for a separate PR, though, as it's not really linked to the feature. The feature's implementation just carries on from having `dataset` loaded in `common.mjs` 😊

Reply:
Yep I'm happy with it not being a blocker, and can address whether both new imports (since the last release) should be moved or not in another PR too

```mjs
import './vendor/polyfills/Element/prototype/dataset.mjs'
import './vendor/polyfills/Element/prototype/closest.mjs'
````